src/docs/asciidoc/configuration.adoc

```diff
@@ -1,9 +1,144 @@
+|===
+
+[[fault-tolerance-configuration]]
+=== Fault Tolerance
+
+[cols="1,5a"]
+|===
+|Configuration API Method|Description
+
+|`setProcessingGuarantee`
+|Processing guarantee for failures. If a cluster member leaves the cluster during a job execution and the job 
+is restarted automatically, it defines the semantics of at which point in the stream the job is resumed from.
+Available values are `EXACTLY_ONCE`, `AT_LEAST_ONCE` and `NONE`. If you choose one of the first two values, 
+distributed snapshotting is for the job. See the {jet-javadoc}/config/ProcessingGuarantee.html[Javadoc] for a detailed
+explanation. Its default value is `NONE`.
+
+|`setSnapshotIntervalMillis`
+|The interval in milliseconds between the completion of the previous snapshot and the start of a new one. It must be a 
+positive value and it is only relevant when the processing guarantee is set as either `EXACTLY_ONCE` or `AT_LEAST_ONCE`.
+Its default value is set to `10000` milliseconds.
+
+|`setAutoScaling`
+|Specifies whether Jet scales the job up or down when a member is added or removed from the cluster. Its default
+value is `true`, i.e., enabled.
+
+|`setSplitBrainProtection`
+|Specifies whether the split-brain protection feature is enabled. When enabled, Jet restarts the job after a topology
+change, only if the cluster quorum is satisfied. See the <<split-brain-protection, Split-Brain Protection section>>.
+|===
+
+[[performance-configuration]]
+=== Performance
+
+[cols="1,5a"]
+|===
+|Configuration API Method|Description
+
+|`setPacketSizeLimit`
+|For a distributed edge, data is sent to a remote member via Hazelcast network packets. Each packet
+is dedicated to the data of a single edge, but may contain any number of data items. This setting
+limits the size of the packet in bytes. Packets should be large enough to drown out any fixed 
+overheads, but small enough to allow good interleaving with other packets. Its default value is `16384`.
+
+|`setQueueSize`
+|The capacity of processor-to-processor concurrent queues. The value is rounded upwards to the
+next power of 2. Its default value is `1024`.
+
+|`setReceiveWindowMultiplier`
+|The scaling factor used by the adaptive receive window sizing function.
+Its default value is `3`.
+|===
+
+[[security-configuration]]
+=== Security
+
+[cols="1,5a"]
+|===
+|Configuration Element|Description
+
+|`ssl`
+|Provides properties to configure TLS for all member to member and member to
+client communications. You need to have the Hazelcast Jet
+Enterprise edition to use this feature. See the <<security, Security chapter>> for more information.
+|===
+
+
+[[monitoring-configuration]]
+=== Monitoring
+
+[cols="1,5a"]
+|===
+|Configuration Method/Property|Description
+
+|`hazelcast.logging.type`
+|The predefined types of built-in logging adapters. See the <<configure-logging, Configure Logging section>>.
+
+|`peek`
+|Used to provide outputs of individual stages when debugging a pipeline.
```

Comment:
Peek isn't a config option, is it? It's a pipeline operator.

Reply:
I thought it would be nice to include since it is used for monitoring purposes?

Reply:
In the left-hand column we should have only config options, but all of them. That's how we intended it. We can mention `peek` somewhere else.

Also, it's an open question whether we want the setter names in the left column or just the property names.

Reply:
Just a comment:
I was planning to include all the configuration elements, i.e, the ones in the declarative (XML) configuration. However, I noticed not all of the configuration options have declarative elements (please confirm). So, I decided to give the configuration related setters/objects/elements(when there is not an explicit setter) in the left hand column and give the available values that can be configured in the right hand column. Jet manual gives the "Programmatic configuration is the simplest way to configure Jet" statement and mostly programmatic examples. Also the left hand column is planned to, kind of, group the configuration values for each feature (security, monitoring, etc.) for the to-be-added-in-the-future configuration values.

I will remove "peek".  But let's agree on the left hand column. For me the current PR looks good in terms of giving the configuration information. Please advise or comment.

Reply:
If something is configurable programmatically but not from XML, then that's an omission we have to fix before the release.

Programmatic config is the simplest to start with, but deployments will most probably have it in XML, otherwise the user would have to write his own Java classes that start the cluster.

Maybe Can can chime in on the naming convention to use for the config elements.

Reply:
@cangencer , @vladoschreiner , could you please comment on the above matter?

Reply:
which element is not available declaratively? This sounds like an oversight if true.

Reply:
@cangencer , As I was looking to hazelcast-jet-full-example.xml, I remember not seeing the declarative equivalents of, for example, setProcessingGuarantee, setSnapshotIntervalMillis, setAutoScaling, or setSplitBrainProtection.

Reply:
Those are job-level so they're configured per job, not on the instance level